### DISCUSSION TITLE
api service with login

### DISCUSSION THREAD

**Discussion by ismailix** [ng-matero/ng-matero#668] [Show and tell]
i'm using the ng-matero theme. When logging in to this theme, in-mem-dataservice.he logs in by connecting to the ts file and getting the username and password information. Instead of https://jsonplaceholder.typicode.com/users how can I log in by entering the email address and password by connecting here or connecting to any api that you know. How can we do this without disturbing the theme you have prepared. thank you. @recca0120 @nzbin

**Comment by nzbin** [comment]
https://github.com/ng-matero/ng-matero/blob/f65318db4908236b4f8b5b711c68b7581a76bed1/src/app/app.config.ts#L53-L57

https://github.com/ng-matero/ng-matero/blob/97e3b3d66393948d3a28a8e9fa7f8e1f619bd368/src/app/core/authentication/login.service.ts#L13-L15

**Comment by ismailix** [comment]
thank you for the reply. 
i started seeing 201 payloads. 
inside my body 

```ts
export class LoginResponseModel {
  authToken :string;
  email: string;
  firstName: string;
  lastName : string;
  organizations: OrganizationModel = new OrganizationModel; 
  phone : string;
  userId : string;
}
```

in this way, even though I see 201 because of the token, I can't go to the dashboard screen. 
What kind of arrangement do I need to make. 
And I also want to keep cookies that are not localstoraged or not. Can you help me?
@nzbin

**Reply by nzbin** [reply]
https://github.com/ng-matero/ng-matero/issues/455#issuecomment-1340987902

You should implement the auth logic and auth guard by yourself.

**Comment by ismailix** [comment]
I want to cancel the token service completely. Simply put, how can we not act? Where should I start and where should I delete? Because I want to use my own login service. @nzbin

**Reply by nzbin** [reply]
1. delete the token interceptor
  https://github.com/ng-matero/ng-matero/blob/main/src/app/core/interceptors/token-interceptor.ts
3. delete the authentication
  https://github.com/ng-matero/ng-matero/tree/main/src/app/core/authentication
